tests/ui/skip_without_default.rs

```diff
@@ -23,7 +23,4 @@ pub struct Opts {
     k: Kind,
 }
 
-fn main() {
-    let opts = Opts::from_args();
-    println!("{:?}", opts);
-}
+fn main() {}
```

Comment:
No span modification here?

Reply:
I'm not sure what you mean by "span modification" here, if it's about the reason why I emptied the `main`'s body - well, before this PR the macro was reporting error by a plain panic. Such panics abort compilation process effectively immediately, so the panic-error is the only error displayed *even if there are other errors*. But now errors are reported by a token stream containing a `compile_error!` invocation and such invocation *does not* abort a compilation right away. Furthermore, any `impl Structopt for Foo` code was not generated, so we're going to have a `Foo does not have "from_args" method` error in addition to our `compile_error!` invocation, and this is not what we test here. So I decided to keep this stuff as clean and simple as possible.

Reply:
Sorry, I was really not clear.

This particular test has the same error message as before, that is
```
error[E0277]: the trait bound `Kind: std::default::Default` is not satisfied
  --> $DIR/skip_without_default.rs:17:10
   |
17 | #[derive(StructOpt, Debug)]
   |          ^^^^^^^^^ the trait `std::default::Default` is not implemented for `Kind`
   |
   = note: required by `std::default::Default::default`

For more information about this error, try `rustc --explain E0277`.
```

Somewhere, in the generated code, we must do something like that:
```rust
k: Default::default(),
```
without any span information. Thus, the error message is at the call site. It would be great to point to the `k` field of the struct. If it imply complicated modification, never mind.

Reply:
Will do

Reply:
Wow, that was really easy, ~2mins total. Apparently my idea of carrying as much as possible span info around proved very useful :)